pkg/cmd/dev/testlogic.go

```diff
@@ -127,7 +131,11 @@ func (d *dev) testlogic(cmd *cobra.Command, commandLine []string) error {
 		return err
 	}
 
-	if !noGen {
+	shouldGen, err := d.shouldGen(ctx)
+	if err != nil {
+		return err
+	}
+	if (shouldGen && !noGen) || forceGen {
```

Comment:
If someone passes `--no-gen` or `--force-gen`, it's a big waste of time to perform the checks with `git`. Instead, short-circuit if one of those flags is passed.